packages/serverpod_client/test/exception_from_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:convert';
+
+import 'package:serverpod_client/serverpod_client.dart';
+import 'package:serverpod_client/src/serverpod_client_shared_private.dart';
+import 'package:test/test.dart';
+
+class TestSerialization extends SerializationManager {}
+
+class TestSerializationImpl extends SerializationManager {
+  @override
+  Object? decodeWithType(String data) {
+    var json = jsonDecode(data);
+
+    if (json['className'] == 'TestException') {
+      return TestException.fromJson(json['data']);
+    }
+
+    throw Exception('Unknown exception type');
+  }
+}
+
+class TestException extends SerializableException {
+  String message;
+
+  TestException(this.message);
+
+  factory TestException.fromJson(Map<String, dynamic> json) {
+    return TestException(json['message']);
+  }
+
+  @override
+  dynamic toJson() {
+    return {
+      'message': message,
+    };
+  }
+}
+
+void main() {
+  group('Given a serializable exception when parsing the exception', () {
+    var exception = getExceptionFrom(
+      data:
+          '{"className": "TestException", "data": {"message": "A custom exception"}}',
+      serializationManager: TestSerializationImpl(),
+      statusCode: 400,
+    );
+
+    test('then a base ServerpodClientException is created', () {
+      expect(exception, isA<TestException>());
+    });
+
+    test('then the message is an unknown error', () {
+      expect(exception.message, 'A custom exception');
+    });
+  });
+
+  group(
+      'Given malformed data with an unknown status code when parsing the exception',
+      () {
+    var exception = getExceptionFrom(
+      data: 'malformed data',
+      serializationManager: TestSerialization(),
+      statusCode: 499,
+    );
+
+    test('then a base ServerpodClientException is created', () {
+      expect(exception, isA<ServerpodClientException>());
+      expect(exception.statusCode, 499);
+    });
+
+    test('then the message is an unknown error', () {
+      expect(exception.message, 'Unknown error, data: malformed data');
+    });
+  });
+
+  group(
+      'Given malformed data with a bad request status code when parsing the exception',
+      () {
+    var exception = getExceptionFrom(
+      data: 'malformed data',
```

Comment:
Suggestion: For completeness, please add malformed data request for the different types as well.

Possibly an opportunity for a group.